src/Query.php

```diff
@@ -36,7 +36,7 @@ class Query extends Expression
      *
      * @var string
      */
-    protected $template_select = 'select[option] [field] [from] [table][join][where][group][having][order][limit]';
+    protected $template_select = '[with]select[option] [field] [from] [table][join][where][group][having][order][limit]';
```

Comment:
Does spaces matter here? Somewhere are, somewhere not, why?

Reply:
No space here because in case `with` is not set then we don't want leading space to show up before `select`.
Basically spaces in template are if it's mandatory to fill particular template tags. Otherwise space is added in rendering process if needed. But I agree - that's not very nice if sometimes we do this and sometimes different.